data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/configuration/CompressionOption.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public enum CompressionOption {
+    NONE("none"),
+    GZIP("gzip"),
+    AUTOMATIC("automatic");
+
+    private static final Map<String, CompressionOption> OPTIONS_MAP = Arrays.stream(CompressionOption.values())
+            .collect(Collectors.toMap(
+                    value -> value.option,
+                    value -> value
+            ));
+
+    private final String option;
+
+    CompressionOption(final String option) {
```

Comment:
Let's mark this as private to enforce using the `fromOptionValue` function by other classes in this package.